src/modules/claims/claims.service.ts

```diff
@@ -582,14 +596,33 @@ export class ClaimsService {
     expirationTimestamp,
   }: IssueClaimOptions): Promise<string | undefined> {
     await this.verifyIssuer(claim.claimType);
-    await this.verifyEnrolmentPrerequisites({ subject, role: claim.claimType });
+
+    const roleDefinition = await this._domainsService.getDefinition({
+      type: NamespaceType.Role,
+      namespace: claim.claimType,
+    });
+
+    await this.verifyEnrolmentPrerequisites({
+      subject,
+      role: claim.claimType,
+      roleDefinition,
+    });
 
     const message: IClaimIssuance = {
       id: v4(),
       requester: subject,
       claimIssuer: [this._signerService.did],
       acceptedBy: this._signerService.did,
     };
+
+    const defaultValidityPeriod = roleDefinition?.['defaultValidityPeriod'];
+    const roleDefinitionExpirationTimestamp = defaultValidityPeriod
```

Comment:
Looks like this variable doesn't really needed. `claimExpirationTimestamp` can be calculated by this formula by replacing undefined to expirationTimestamp || defaultExpirationTimestamp

Reply:
Yeah I know, I make for more clarity in code